src/Nautilus/Asset/Service/ServiceBuilder.ts

```diff
@@ -172,35 +172,41 @@ export class ServiceBuilder<
     return this
   }
 
-  addTrustedAlgorithm(algorithm: PublisherTrustedAlgorithm) {
+  addTrustedAlgorithms(algorithms: PublisherTrustedAlgorithm[]) {
     if (this.service.type !== 'compute') {
       throw new Error('Illegal operation, asset is not a compute asset!')
     }
 
-    if (!this.service.compute.publisherTrustedAlgorithms) {
-      this.service.compute.publisherTrustedAlgorithms = [algorithm]
-      return this
+    // Check if the algorithms array is empty
+    if (algorithms.length === 0) {
```

Comment:
issue: could be undefined
```suggestion
    if (algorithms?.length === 0) {
```